quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/ClassLoaderPropertyMappers.java

```diff
@@ -27,14 +34,27 @@ public static PropertyMapper[] getMappers() {
     private static Optional<String> resolveIgnoredArtifacts(Optional<String> value, ConfigSourceInterceptorContext context) {
         if (Environment.isRebuildCheck() || Environment.isRebuild()) {
             Profile profile = getCurrentOrCreateFeatureProfile();
+            Set<String> ignoredArtifacts = new HashSet<>();
 
             if (profile.getFeatures().get(Feature.FIPS)) {
-                return Optional.of(
-                        "org.bouncycastle:bcprov-jdk15on,org.bouncycastle:bcpkix-jdk15on,org.bouncycastle:bcutil-jdk15on,org.keycloak:keycloak-crypto-default");
+                ignoredArtifacts.addAll(List.of(
+                        "org.bouncycastle:bcprov-jdk15on", "org.bouncycastle:bcpkix-jdk15on", "org.bouncycastle:bcutil-jdk15on", "org.keycloak:keycloak-crypto-default"));
+            } else {
+                ignoredArtifacts.addAll(List.of(
+                        "org.keycloak:keycloak-crypto-fips1402", "org.bouncycastle:bc-fips", "org.bouncycastle:bctls-fips", "org.bouncycastle:bcpkix-fips"));
+            }
+
+            Optional<String> storage = Configuration.getOptionalValue(
+                    MicroProfileConfigProvider.NS_KEYCLOAK_PREFIX + StorageOptions.STORAGE.getKey());
+
+            if (storage.isEmpty()) {
+                ignoredArtifacts.add("org.keycloak:keycloak-model-map-jpa");
+                ignoredArtifacts.add("org.keycloak:keycloak-model-map-hot-rod");
+                ignoredArtifacts.add("org.keycloak:keycloak-model-map");
+                ignoredArtifacts.add("org.keycloak:keycloak-model-map-file");
```

Comment:
IMHO, would be good to manage the ignored artifacts separately for better maintainability. As we will consider removing also irrelevant JDBC drivers and metrics/health artifacts, I'd propose to improve it little bit with this PR: https://github.com/pedroigor/keycloak/pull/12

@pedroigor @vmuzikar WDYT?

Reply:
Works for me as a follow-up.

Reply:
+1